src/compiler/crystal/macros/methods.cr

```diff
@@ -965,6 +965,11 @@ module Crystal
         interpret_check_args { @of.try(&.key) || Nop.new }
       when "of_value"
         interpret_check_args { @of.try(&.value) || Nop.new }
+      when "has_key?"
+        interpret_check_args do |key|
+          entry = entries.find &.key.==(key)
+          BoolLiteral.new(entry != nil)
```

Comment:
**suggestion:** `entry.nil?` is more idiomatic than `entry == nil`. `#==` could theoretically be overridden, `#nil?` cannot.
```suggestion
          BoolLiteral.new(!entry.nil?)
```